src/librustdoc/doctest.rs

```diff
@@ -468,6 +472,9 @@ fn run_test(
                 return Err(TestFailure::UnexpectedRunPass);
             } else if !should_panic && !out.status.success() {
                 return Err(TestFailure::ExecutionFailure(out));
+            } else if options.nocapture {
+                io::stdout().write_all(&out.stdout).expect("failed to write stdout");
+                io::stderr().write_all(&out.stderr).expect("failed to write stderr");
```

Comment:
I have two remaining questions:

- Is it correct for this to only print the output when the test passes? `--nocapture` sounds to me like it won't capture the output at all. What does libtest do?
- This will still appear to capture the output when a test hangs, and it will also delay when the output is shown for slow tests.

The overarching question I have is: Why did you decide to still capture the output, but then print it at the end? I would think it would give less surprising results if we just didn't capture the output in the first place.

Reply:
This is a good point. I don't remember the reason but I guess it was around "compile_fail", however it doesn't make sense to care about it so I'll simply prevent to capture it.

Reply:
Done, the output is now free! \o/